### PR TITLE
[GEOS-11075] IAU authority, planetary CRS support

### PR DESCRIPTION
**Pull request by aaime**
[GEOS-11075](https://osgeo-org.atlassian.net/browse/GEOS-11075)

See ticket, and discussion on list. This is pure packaging of the gt-iau-wkt module, with no GeoServer specific additions.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: doc/en/user/source/extensions/iau/install.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#. Extract the archive and copy the contents into the GeoServer :file:`WEB-INF/lib` directory.
+
+#. Restart GeoServer.
+
+Verify Installation
+^^^^^^^^^^^^^^^^^^^
+
+To verify that the extension was installed successfully:
+
+#. Try to set up a layer (e.g., shapefile) with a IAU CRS (e.g., ``IAU:49900``, Mars (2015) - Sphere / Ocentric)
```
Comment (jodygarnett) [inline doc/en/user/source/extensions/iau/install.rst:21]: I would of checked with the SRS demo page :)

> http:/localhost:8080/geoserver/web/wicket/bookmarkable/org.geoserver.web.demo.SRSDescriptionPage?3&code=IAU:49900

Easier test?

File: src/release/ext-iau.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+       <outputDirectory></outputDirectory>
+       <includes>
+          <include>gt-iau-wkt*.jar</include>
+       </includes>
+    </fileSet>
+    <fileSet>
+      <directory>release/target/html/licenses</directory>
+      <outputDirectory></outputDirectory>
+      <includes>
+         <include>LGPL.html</include>
```
Comment (jodygarnett) [inline src/release/ext-iau.xml:26]: Do the definitions have a data license at all?

**Review by jodygarnett — APPROVED** [review]
PR looks good, some feedback for your consideration.

File: doc/en/user/source/extensions/iau/install.rst (hunk 1 above)
Comment (aaime) [reply doc/en/user/source/extensions/iau/install.rst:21]: Not implemented yet in the SRS page, it's still bound to the EPSG authority

File: src/release/ext-iau.xml (hunk 2 above)
Comment (aaime) [reply src/release/ext-iau.xml:26]: The jar contains also Java code, the SPI needs a dedicated subclass of the property based authority to work, even if the overrides are minimal (hence LGPL). I have no idea about the definitions.

File: doc/en/user/source/extensions/iau/install.rst [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+#. The layer CRS should recognize the CRS and may already provide ``IAU:49900`` in the declared CRS field (or not, depending on whether the original CRS had an authority and code indication, or just a basic definition)
+#. Even if not indicated automatically, it should be possible to enter a IAU code and it is accepted as a valid one.
```
Comment (bradh) [inline doc/en/user/source/extensions/iau/install.rst:23]:
```suggestion
#. Even if not indicated automatically, it should be possible to enter a IAU code and have it accepted as valid.
```

File: src/release/extensions/iau/iau-readme.txt [hunk 4]
```diff
@@ -0,0 +1,24 @@
+GeoServer IAU authority readme
+------------------------------
+
+This package contains the International Astronominal Unit CRS database,
```
Comment (bradh) [inline src/release/extensions/iau/iau-readme.txt:4]:
```suggestion
This package contains the International Astronomical Union CRS database,
```

File: src/release/extensions/iau/iau-readme.txt [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+Please report any bugs with jira (https://osgeo-org.atlassian.net/projects/GEOS).
+
+Any other issues can be discussed on the mailing list (http://lists.sourceforge.net/lists/listinfo/geoserver-users).
+
+Contains the following data stores:
+
+INSTALLATION
+
+1. Copy included gt2-iau-wak jar included to your
```
Comment (bradh) [inline src/release/extensions/iau/iau-readme.txt:15]:
```suggestion
1. Copy included gt-iau-wkt jar included to your 
```

**Review by bradh — CHANGES_REQUESTED** [review]

**Review by bradh — APPROVED** [review]
Lgtm

**Pull request merged**